demo/src/app/components/typeahead/overview/sections/basic-usage/template.html

```diff
@@ -0,0 +1,47 @@
+<p>
+  To set up a Typeahead instance, you need to pass an initialization function to it. This function must return an Observable spawning lists of options: this is how you control the list of displayed options.
+</p>
+
+<p>
+  Pass it that way on the template side:
+</p>
+
+<ngbd-code [lang]="snippets.observableSetup.template.language" [code]="snippets.observableSetup.template.code" [highlightedLines]="snippets.observableSetup.template.highlightedLines"></ngbd-code>
+
+<p>
+  And define it that way on the component side:
+</p>
+
+<ngbd-code [lang]="snippets.observableSetup.component.language" [code]="snippets.observableSetup.component.code" [highlightedLines]="snippets.observableSetup.component.highlightedLines"></ngbd-code>
+
+
+
+<h5>The basic Observable chain</h5>
+
+<p>
+  In this basic example we will simply use a statically defined and synchronously available set of data — color names:
+</p>
+
+<ngbd-code [lang]="snippets.data.language" [code]="snippets.data.code" [highlightedLines]="snippets.data.highlightedLines"></ngbd-code>
+
+<p>
+  The function used to provide the Observable to Typeahead receives a single argument: another Observable. It spawns:
+</p>
+<ul>
+  <li>what: strings corresponding to the current content of the input</li>
+  <li>when: on the <code>input</code> event of the input</li>
+</ul>
+
+<p>
+  You can therefore use this input as a search term and apply filtering for example:
+</p>
+
+<ngbd-code [lang]="snippets.observable.mapping.language" [code]="snippets.observable.mapping.code" [highlightedLines]="snippets.observable.mapping.highlightedLines"></ngbd-code>
```

Comment:
Because the initial abstraction is poorly designed according to me. So I don't break things, but I make it work in a way that is more logical.

See it that way: with the current abstraction, the code and the language of the snippet are defined in two separate locations, while they are intrinsically linked. So why defining it in two separate locations??

What I would have liked to do is just pass a snippet object which has all the information in one place. Even for line highlighting, it should rather sit along with the code of the snippet otherwise you'll jump between two files to update things.

Something like: 

```html
<ngbd-code [snippet]="path.to.my.snippet"></ngbd-code>
```

Reply:
> you'll jump between two files to update things

That was my point. Firstly, you never change snippet type. Secondly, without highlighted lines it is still fine the way it is.

- it is out of scope of typeahead overview
- there is a #3111 if you want to continue discussion about `<ngbd-code>` specifically

Reply:
It happened to me that I was inserting some snippet somewhere, and wanted to change the snippet afterwards, sometimes with different language. It was easier to blindly change the "path" to the snippet rather than having to check its type, and update the language property specifically. And splitting the two remains illogical.

Since changing the API of the code component is not part of this PR, I can at least write the documentation in a way that appears more logical? After all, it just means: "insert this snippet there", you don't care which language it is when reading the template. The, you check the snippet and there you have all the information.